spark2/src/test/java/org/apache/iceberg/spark/source/TestSparkTableUtil.java

```diff
@@ -298,4 +309,69 @@ public void testImportWithNameMappingForVectorizedParquetReader() throws Excepti
 
     Assert.assertEquals(expected.stream().map(SimpleRecord::getData).collect(Collectors.toList()), actual);
   }
+
+  @Test
+  public void testAvroReaderWithNameMapping() throws IOException {
+    File avroFile = temp.newFile();
+    org.apache.avro.Schema avroSchema = SchemaBuilder.record("TestRecord")
+        .namespace("org.apache.iceberg.spark.data")
+        .fields()
+        .requiredInt("id")
+        .requiredString("name")
+        .endRecord();
+
+    GenericRecord record1 = new GenericData.Record(avroSchema);
+    record1.put("id", 1);
+    record1.put("name", "Bob");
+
+    GenericRecord record2 = new GenericData.Record(avroSchema);
+    record2.put("id", 2);
+    record2.put("name", "Alice");
+
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(avroSchema);
+    DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+
+    dataFileWriter.create(avroSchema, avroFile);
+    dataFileWriter.append(record1);
+    dataFileWriter.append(record2);
+    dataFileWriter.close();
+
+    DataFile avroDataFile = DataFiles.builder(PartitionSpec.unpartitioned())
+        .withFormat("avro")
+        .withFileSizeInBytes(avroFile.length())
+        .withPath(avroFile.getAbsolutePath())
+        .withRecordCount(2)
+        .build();
+
+    Schema filteredSchema = new Schema(
+        required(1, "name", Types.StringType.get())
+    );
+
+    NameMapping nameMapping = MappingUtil.create(filteredSchema);
+
+    Table table = catalog.createTable(
+        org.apache.iceberg.catalog.TableIdentifier.of(DB_NAME, "avro_table"),
+        filteredSchema,
+        PartitionSpec.unpartitioned());
+
+    table.updateProperties()
+        .set(DEFAULT_NAME_MAPPING, NameMappingParser.toJson(nameMapping))
+        .set(DEFAULT_FILE_FORMAT, "avro")
```

Comment:
FYI, this isn't necessary because the files added to a table don't have to match the format.